Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,459 @@
+                    CommitMessage = $prBranchName
+                    BranchName = $prBranchName
+                }
+                $repo | Set-GitHubContent @setContentParms
+                $pullRequest = $prBranch | New-GitHubPullRequest -Title $prBranchName -Head $prBranchName -Base $mainBranchName
+
+                $newGitHubCodespaceParms = @{
+                    PullRequest = $pullRequest.number
+                    RepositoryId = $repo.Id
+                    Wait = $true
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.pulls_url | Should -Be $pullRequest.url
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with all possible settings' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    # ClientIp = 'TODO ???? - should be instead of rather than in addition to Geo, perhaps add some param validation to the function'
+                    # DevContainerPath = 'Will add to test in the future when Get-GitHubDevContainer is implemented and the test repo includes one'
+                    DisplayName = 'PowerShellForGitHub pester test'
+                    Geo = 'UsWest'
+                    Machine = 'basicLinux32gb'
+                    NoMultipleRepoPermissions = $true # Not sure how to assert this, but this proves it accepts the switch without error
+                    IdleRetentionPeriodMinutes = 10
+                    TimeoutMinutes = 5
+                    # WorkingDirectory = 'TODO ???? - not sure how to handle this'
+                }
+                $codespace = $repo | New-GitHubCodespace @newGitHubCodespaceParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                # $codespace.devcontainer_path | Should -Be
+                $codespace.display_name | Should -Be $newGitHubCodespaceParms.DisplayName
+                $codespace.idle_timeout_minutes | Should -Be $newGitHubCodespaceParms.TimeoutMinutes
+                $codespace.location | Should -Match 'WestUs'
```

Comment:
I disagree.  We request Geo=UsWest, and the returned location should reflect that.  It should not return a EastUs location.  And, there is no Geo property in the output.

Reply:
I missed both that you were setting the `Geo` in this test, as well as the fact that `Geo` isn't in returned response.  I'm ok with you keeping this then, but it may be worth just adding a comment to 323 that references the fact that this is the translated location based on the specified `USWest` Geo.

Reply:
Comments added.